azure/defaults.go

```diff
@@ -299,21 +309,37 @@ func GetDefaultWindowsImage(k8sVersion string) (*infrav1.Image, error) {
 }
 
 // GetBootstrappingVMExtension returns the CAPZ Bootstrapping VM extension.
-// The CAPZ Bootstrapping extension is a simple clone of https://github.com/Azure/custom-script-extension-linux which allows running arbitrary scripts on the VM.
+// The CAPZ Bootstrapping extension is a simple clone of https://github.com/Azure/custom-script-extension-linux for Linux or
+// https://docs.microsoft.com/en-us/azure/virtual-machines/extensions/custom-script-windows for Windows.
+// This extension allows running arbitrary scripts on the VM.
 // Its role is to detect and report Kubernetes bootstrap failure or success.
-func GetBootstrappingVMExtension(osType string, cloud string) (name, publisher, version string) {
-	// currently, the bootstrap extension is only available for Linux and in AzurePublicCloud.
-	if osType == "Linux" && cloud == azure.PublicCloud.Name {
-		return "CAPZ.Linux.Bootstrapping", "Microsoft.Azure.ContainerUpstream", "1.0"
+func GetBootstrappingVMExtension(osType string, cloud string, vmName string) *ExtensionSpec {
+	// currently, the bootstrap extension is only available in AzurePublicCloud.
+	if osType == LinuxOS && cloud == azure.PublicCloud.Name {
+		// The command checks for the existence of the bootstrapSentinelFile on the machine, with retries and sleep between retries.
+		return &ExtensionSpec{
+			Name:      "CAPZ.Linux.Bootstrapping",
+			VMName:    vmName,
+			Publisher: "Microsoft.Azure.ContainerUpstream",
+			Version:   "1.0",
+			ProtectedSettings: map[string]string{
+				"commandToExecute": LinuxBootstrapExtensionCommand,
+			},
+		}
+	} else if osType == WindowsOS && cloud == azure.PublicCloud.Name {
+		// This command always returns success (for now).
```

Comment:
It should fail with the powershell if the command isn't there?

Reply:
good catch - this comment was stay (initially I was testing with `cmd.exe /c exit /b 0` for the extension command which did always return 0/success)

Reply:
Fixed